### ISSUE TITLE
[BUG]  When translating, the translation result will keep adding text.

### ISSUE THREAD

**Issue by codeAndxv** [niedev/RTranslator#81]
Thank you for wanting to report a bug! But before starting, ensure to check if this bug report respects the following requirements:
- [x] It is written in English (I can translate what you say, but issues written in English are easier to read for the other users).
- [x] There is not already a similar bug report among the open issues (if this is the case you can leave a comment describing your problem in more detail in that already opened issue)

**Describe the bug**
When I type "你好啊世界如何看待目前的经济局势失业很严重大家都找不到工作未来会变的好吗", it is translated from Chinese to English. The translation results will always append text, just like the one I took in the screenshot. I'm using the latest v2.0.0 commit code
,[6bf51c1e2f70](https://github.com/niedev/RTranslator/commit/6bf51c1e2f703096ec13da8b57d0fcbfc835b6a1)

**Screenshots**

<img src="https://github.com/user-attachments/assets/4a4174a3-2d8d-4d5b-8214-85ad7382662b" alt="Image 1" width="240" style="display: inline-block; margin-right: 10px;">
<img src="https://github.com/user-attachments/assets/e88554d5-b992-4afc-805a-542c4c8007e4" alt="Image 3" width="240" style="display: inline-block;">
<img src="https://github.com/user-attachments/assets/4a77c99d-b532-4a9b-a290-ab92e54157d0" alt="Image 2" width="240" style="display: inline-block; margin-right: 10px;">

**Smartphone (please complete the following information):**
 - Device: xiaomi 12
 - Android OS version: Android 13
 - TTS used: No
 - Version of the app: 2.1.1

**Additional context**

**Comment by niedev** [comment]
Thank you for reporting this bug, I will investigate on this.

**Comment by codeAndxv** [comment]
> Thank you for reporting this bug, I will investigate on this.

Hi, is this problem difficult to solve, and can you estimate how long it will take to resolve it? Because this issue occurs frequently, it's quite troublesome for me. Thanks!   By the way, if you need help, I know some Java and Android.

**Comment by niedev** [comment]
I did some testing and the problem seems to be due to the quantization of the models (and greater for Chinese), which would make it quite difficult to solve. However I tried adding a period at the end of the sentence that gave you problems and in that case the problem goes away. For now you could try to always add a period at the end of input sentences, let me know if this solves or reduces the problem, in that case I will make an update in which the app will automatically add a period at the end of sentences when needed.

Thanks again.

<img align="left" width="400" src="https://github.com/user-attachments/assets/c5f840f3-85ce-4793-bce4-3bf02e36c606">

**Comment by codeAndxv** [comment]
In addition to adding a full stop at the end, and forcibly breaking the entered text.

I tried to add the  java  code snapper below in [this place](https://github.com/niedev/RTranslator/blob/v2.00/app/src/main/java/nie/translator/rtranslator/voice_translation/neural_networks/translation/Translator.java#L768).  Without modifying the model, this problem can be solved. 
```        
if (j > 2 * input.getInputIDs().length) {
      android.util.Log.i("decoder", "Reach max tokenId count: " + 2 * input.getInputIDs().length + ", maybe occur error, break while.");
      break;
}
```  

But the root cause is a model problem.

**Comment by niedev** [comment]
@codeAndxv Thank you for your input, this would definitely help. I'll try this change and others next week (I started a new job so I don't have much time, but I should be able to get around to it).

**Comment by niedev** [comment]
The [new release](https://github.com/niedev/RTranslator/releases/tag/2.1.2) that should mitigate this problem is out 🚀, let me know if the problem occurs less.

**Comment by codeAndxv** [comment]
It look fine,  Thanks.

**Comment by zhl111** [comment]
![SmartSelect_20241103_173748_RTranslator](https://github.com/user-attachments/assets/9fdd8cba-0e58-46f0-8107-a894e3ad3eb5)
After I said one sentence, everything turned gray and nothing clicked except the return key.

**Comment by niedev** [comment]
@zhl111 have you tried RTranslator 2.1.2?

**Comment by zhl111** [comment]
Yes, this is the version I use. My mobile phone is Samsung s23ultra, is it related to the system of the mobile phone?

**Comment by niedev** [comment]
@zhl111 Ok, I don't think this is due to the system of the phone, but it can be due to the tts you use. Everything turn gray (and stay gray) after the transcription appear or before? And if you turn off the TTS (from the top right speaker button in the WalkieTalkie mode) the problem goes away?

**Comment by zhl111** [comment]
It's all right

**Comment by niedev** [comment]
@zhl111 what tts are you using?

**Comment by zhl111** [comment]
google tts

**Comment by niedev** [comment]
@zhl111 That's strange, when everything turn gray, it stay gray forever or it return green after the sentence is spoken?

**Comment by zhl111** [comment]
Before, all the buttons were gray, now they turn green after the sentence is finished

**Comment by niedev** [comment]
@zhl111 Ok, before the bug we discussed was the problem, now the app behaves normally. In WalkieTalkie mode while the app processes the translation and the tts speaks the microphone deactivates because otherwise, the app would recognize the audio of the tts, translating in a loop (the Conversation mode instead, using Bluetooth headphones, never interrupts the microphones because with headphones you obviously don't have this problem).
